gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,442 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+from astropy.extern.six.moves import range
+from astropy.extern.six import iteritems
+
+import logging
+log = logging.getLogger(__name__)
+
+__all__ = ['fc_find_acceptance_region_gauss',
+           'fc_find_acceptance_region_poisson',
+           'fc_construct_acceptance_intervals_pdfs',
+           'fc_get_upper_and_lower_limit',
+           'fc_fix_upper_and_lower_limit',
+           'fc_find_limit',
+           'fc_find_average_upper_limit',
+           'fc_construct_acceptance_intervals',
+          ]
+
+
+def fc_find_acceptance_region_gauss(mu, sigma, x_bins, alpha):
+    """Analytically find the acceptance region (x_min, x_max) for Gaussian with
+       boundary at the origin, such that int_x_min^x_max P(x|mu)dx = alpha
+
+    Parameters
+    ----------
+    mu : double
+        Mean of the Gaussian
+    sigma : double
+        Width of the Gaussian
+    x_bins : array-like
+        Bins in x
+    alpha : double
+        Desired confidence level
+
+    Returns
+    -------
+    (x_min, x_max) : tuple of floats
+        Acceptance interval
+    """
+
+    from scipy import stats
+
+    dist = stats.norm(loc=mu, scale=sigma)
+
+    x_bin_width = x_bins[1] - x_bins[0]
+
+    p = []
+    r = []
+
+    for x in x_bins:
+        p.append(dist.pdf(x)*x_bin_width)
+        # This is the formula from the FC paper
+        if mu == 0 and sigma == 1:
+            if x < 0:
+                r.append(np.exp(mu*(x-mu*0.5)))
+            else:
+                r.append(np.exp(-0.5*np.power((x-mu),2)))
+        # This is the more general formula
+        else:
+            # Implementing the boundary condition at zero
+            muBest     = max(0, x)
+            probMuBest = stats.norm.pdf(x, loc=muBest, scale=sigma)
+            if probMuBest == 0.0:
+                r.append(0.0);
```

Comment:
@dlennarz – I don't have time to re-read the FC paper and do a review of your implementation whether the method is OK. So all I can do here before this goes in is superficial comments (sorry!) and one thing I watch out for is test coverage. Test coverage is already very good (much better than for most parts of Gammapy), still, I'll point out the few lines that aren't covered and ask you if it's possible / reasonable to add a test case that exercises them.

This line is one of those that aren't tested:
```python
r.append(0.0);
```